src/games/strategy/triplea/ui/screen/UnitsDrawer.java

```diff
@@ -24,86 +25,92 @@
+  private static boolean drawUnitNation = true;
   public UnitsDrawer(final int count, final String unitType, final String playerName, final Point placementPoint,
       final int damaged, final int bombingUnitDamage, final boolean disabled, final boolean overflow,
       final String territoryName, final IUIContext uiContext2) {
-    m_count = count;
-    m_unitType = unitType;
-    m_playerName = playerName;
-    m_placementPoint = placementPoint;
-    m_damaged = damaged;
-    m_bombingUnitDamage = bombingUnitDamage;
-    m_disabled = disabled;
-    m_overflow = overflow;
-    m_territoryName = territoryName;
-    m_uiContext = uiContext2;
+    this.count = count;
+    this.unitType = unitType;
+    this.playerName = playerName;
+    this.placementPoint = placementPoint;
+    this.damaged = damaged;
+    this.bombingUnitDamage = bombingUnitDamage;
+    this.disabled = disabled;
+    this.overflow = overflow;
+    this.territoryName = territoryName;
+    this.uiContext = uiContext2;
   }
 
-  public void prepare() {}
-
   public Point getPlacementPoint() {
-    return m_placementPoint;
+    return placementPoint;
   }
 
   public String getPlayer() {
-    return m_playerName;
+    return playerName;
   }
 
   @Override
   public void draw(final Rectangle bounds, final GameData data, final Graphics2D graphics, final MapData mapData,
       final AffineTransform unscaled, final AffineTransform scaled) {
-    if (m_overflow) {
+    if (overflow) {
       graphics.setColor(Color.BLACK);
-      graphics.fillRect(m_placementPoint.x - bounds.x - 2,
-          m_placementPoint.y - bounds.y + m_uiContext.getUnitImageFactory().getUnitImageHeight(),
-          m_uiContext.getUnitImageFactory().getUnitImageWidth() + 2, 3);
+      graphics.fillRect(placementPoint.x - bounds.x - 2,
+          placementPoint.y - bounds.y + uiContext.getUnitImageFactory().getUnitImageHeight(),
+          uiContext.getUnitImageFactory().getUnitImageWidth() + 2, 3);
     }
-    final UnitType type = data.getUnitTypeList().getUnitType(m_unitType);
+    final UnitType type = data.getUnitTypeList().getUnitType(unitType);
     if (type == null) {
-      throw new IllegalStateException("Type not found:" + m_unitType);
+      throw new IllegalStateException("Type not found:" + unitType);
+    }
+    final PlayerID owner = data.getPlayerList().getPlayerID(playerName);
+    final Image img = uiContext.getUnitImageFactory().getImage(type, owner, data,
+        damaged > 0 || bombingUnitDamage > 0, disabled);
+    graphics.drawImage(img, placementPoint.x - bounds.x, placementPoint.y - bounds.y, null);
+    if(!STATIC_UNITS.contains(type.getName()) && drawUnitNation){
+      final Image flag = uiContext.getFlagImageFactory().getSmallFlag(owner);
+      final int xoffset = img.getWidth(null) - flag.getWidth(null);//If someone wants to put more afford in this feature he could add an algorithm calculation the right-bottom corner
```

Comment:
be more precise with the TODO comment if you could. It does not necessarily make a lot of sense.